config/jobs/ci-infra/ci-infra-automate-testgrid.yaml

```diff
@@ -1,60 +1,56 @@
-presubmits: 
+presubmits:
   gardener/ci-infra:
-  - name: pull-ci-infra-validate-testgrid-config
+  - name: pull-ci-infra-check-testgrid-config
     cluster: gardener-prow-build
-    run_if_changed: '^(config\/jobs\/.*.yaml)|(config\/prow\/gardener-testgrid.yaml)$'
     branches:
     - ^master$
+    run_if_changed: '^(config\/jobs\/.*.yaml)|(config\/testgrids\/.*.yaml)$'
+    decorate: true
     annotations:
       testgrid-create-test-group: "false"
-    decorate: true
     spec:
       containers:
-      - image: "gcr.io/k8s-prow/transfigure:v20220523-6e12e42c76"
+      - image: gcr.io/k8s-prow/configurator:v20220618-82a6661467
         command:
-        - "/ko-app/transfigure"
+        - /ko-app/configurator
         args:
-        - "test"
-        - "config/prow/config.yaml"
-        - "config/jobs"
-        - "config/prow/gardener-testgrid.yaml"
-        - "gardener"
-        - "kubernetes-test-infra"
+        - --yaml=config/testgrids/config.yaml
+        - --default=config/testgrids/default.yaml
+        - --prow-config=config/prow/config.yaml
+        - --prow-job-config=config/jobs
+        - --prowjob-url-prefix=https://github.com/gardener/ci-infra/tree/master/config/jobs
+        - --update-description
+        - --validate-config-file
+        - --oneshot
         resources:
           requests:
-            memory: 1.5Gi
-            cpu: 1
+            memory: "1Gi"
 postsubmits:
   gardener/ci-infra:
-  - name: post-ci-infra-update-testgrid-config
-    cluster: gardener-prow-trusted
-    run_if_changed: '^(config\/jobs\/.*.yaml)|(config\/prow\/gardener-testgrid.yaml)$'
+  - name: post-ci-infra-upload-testgrid-config
+    cluster: gardener-prow-build
     branches:
     - ^master$
+    max_concurrency: 1
+    run_if_changed: '^(config\/jobs\/.*.yaml)|(config\/testgrids\/.*.yaml)$'
+    decorate: true
     annotations:
       testgrid-create-test-group: "false"
-    decorate: true
     spec:
+      serviceAccountName: gardener-prow-storage
```

Comment:
I don't think this will work for us, as our `gardener-prow-storage` service account is not bound to a GCP service account like in [k/test-infra](https://github.com/kubernetes/test-infra/blob/c48f614070863b564e810e3f78fce865d598cfee/config/prow/cluster/trusted_serviceaccounts.yaml#L14).

We probably need to mount the secret and specify `--gcp-service-account`.

Reply:
Added the option. Apparently, I was stuck in the example config 😅 

The secret should be mounted already because of `decorate: true`
- [Secret path](https://github.com/kubernetes/test-infra/blob/5ca3a768c08715f87283ece56bc2f13ac62827cf/prow/pod-utils/decorate/podspec.go#L57-L58)
- Adding [volume mount](https://github.com/kubernetes/test-infra/blob/5ca3a768c08715f87283ece56bc2f13ac62827cf/prow/pod-utils/decorate/podspec.go#L598-L609)